tests/test_features.py

```diff
@@ -92,60 +92,61 @@ def test_stone_features(self):
             X.X......
             .........''' + EMPTY_ROW * 7))
 
+        all_zeros = np.zeros([go.N, go.N])
         for i in range(10, 16):
-            self.assertEqualNPArray(f[:, :, i], np.zeros([go.N, go.N]))
+            self.assertEqualNPArray(all_zeros, f[:, :, i])
 
     def test_stone_color_feature(self):
         f = features.stone_color_feature(TEST_POSITION)
-        self.assertEqual(f.shape, (9, 9, 3))
+        self.assertEqual((9, 9, 3), f.shape)
         # plane 0 is B
-        self.assertEqual(f[0, 1, 0], 1)
-        self.assertEqual(f[0, 1, 1], 0)
+        self.assertEqual(1, f[0, 1, 0])
+        self.assertEqual(0, f[0, 1, 1])
         # plane 1 is W
-        self.assertEqual(f[0, 8, 1], 1)
-        self.assertEqual(f[0, 8, 0], 0)
+        self.assertEqual(0, f[0, 8, 0])
+        self.assertEqual(1, f[0, 8, 1])
         # plane 2 is empty
-        self.assertEqual(f[0, 5, 2], 1)
-        self.assertEqual(f[0, 5, 1], 0)
+        self.assertEqual(1, f[0, 5, 2])                                 # < WHAAAAAAATTTT
+        self.assertEqual(0, f[0, 5, 1])
 
     def test_liberty_feature(self):
         f = features.liberty_feature(TEST_POSITION)
         self.assertEqual(f.shape, (9, 9, features.liberty_feature.planes))
 
-        self.assertEqual(f[0, 0, 0], 0)
+        self.assertEqual(0, f[0, 0, 0])
         # the stone at 0, 1 has 3 liberties.
-        self.assertEqual(f[0, 1, 2], 1)
-        self.assertEqual(f[0, 1, 4], 0)
+        self.assertEqual(1, f[0, 1, 2])
+        self.assertEqual(0, f[0, 1, 4])
         # the group at 0, 7 has 3 liberties
-        self.assertEqual(f[0, 7, 2], 1)
-        self.assertEqual(f[0, 8, 2], 1)
+        self.assertEqual(1, f[0, 7, 2])
+        self.assertEqual(1, f[0, 8, 2])
         # the group at 1, 0 has 18 liberties
-        self.assertEqual(f[1, 0, 7], 1)
+        self.assertEqual(1, f[1, 0, 7])
 
     def test_recent_moves_feature(self):
         f = features.recent_move_feature(TEST_POSITION)
         self.assertEqual(f.shape, (9, 9, features.recent_move_feature.planes))
         # most recent move at (1, 0)
-        self.assertEqual(f[1, 0, 0], 1)
-        self.assertEqual(f[1, 0, 3], 0)
+        self.assertEqual(1, f[1, 0, 0])
+        self.assertEqual(0, f[1, 0, 3])                             # < WHAT IS THIS TESTING?
```

Comment:
the history planes?

Reply:
testing that the most recent move played doesn't exist on history planes before it's been played.

Reply:
My understanding is that ordering of the planes is?
```
012345
BWBWBW
```
So you're checking that 1,0 is B most recently, and not W two turns ago?

Reply:
yeah it looks like it should be s/3/2 to me

Reply:
hasn't changed since our fork:
https://github.com/brilee/MuGo/blob/master/tests/test_features.py#L77

Reply:
Nope, I'm just wrong about how these features are laid out. removing comments